### PR TITLE
[ci]: fix failing tests.

### PR DESCRIPTION
**Pull request by appetrosyan**
## Description of change

Fixed failing CI pipelines in the `iroha2-dev` to `iroha2` merge pipeline. 

## Issue
None

## Benefits

Working CI pipeline

## Possible drawbacks 

None

### CODE REVIEW

File: permissions_validators/Cargo.toml [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
-iroha_macro = { version = "=2.0.0-pre.1", path = "../macro" }
+iroha_core = { version = "=2.0.0-pre-rc.2", path = "../core", default-features = false }
+iroha_data_model = { version = "=2.0.0-pre-rc.2", path = "../data_model", default-features = false }
+iroha_macro = { version = "=2.0.0-pre-rc.2", path = "../macro" }
 
 once_cell = "1.8.0"
+
+[dev-dependencies]
+iroha_data_model = { version = "=2.0.0-pre-rc.2", path = "../data_model", default-features = false, features = ["roles"] }
+iroha_core = { version = "=2.0.0-pre-rc.2", path = "../core", default-features = false, features = ["roles"] }
```
Comment (s8sato) [inline permissions_validators/Cargo.toml:21]: Add a newline

File: permissions_validators/src/private_blockchain/query.rs [hunk 2]
```diff
@@ -226,6 +226,8 @@ impl<W: WorldTrait> IsAllowed<W, QueryBox> for OnlyAccountsDomain {
                     ))
                 }
             }
+            #[cfg(not(feature = "roles"))]
+            _ => Err("Unable to compile with tests enabled, but without \"roles\", because it's a dev-dependency of `iroha_data_model`, and due to https://github.com/rust-lang/cargo/issues/6915".to_owned())
```
Comment (s8sato) [inline permissions_validators/src/private_blockchain/query.rs:230]: Is this case reachable?

Comment (appetrosyan) [reply permissions_validators/src/private_blockchain/query.rs:230]: No. But without this case, it won't compile. 

Basically, `#[cfg(test)]` pulls in all the `dev` dependencies. There are a few tests in `data_model` that require `roles` so when you compile `permissions_validators` in `test` mode, it compiles `iroha_data_model` with its `iroha_core` `"roles"` dependency, and so you need to have this exhaustive list of query values. 

Once we can specify feature dependencies in test mode, (the linked issue) we should remove this and make sure that when we compile without `roles`, then all crates are compiled without them.

Comment (s8sato) [reply permissions_validators/src/private_blockchain/query.rs:230]: I see, thanks

**Review by s8sato — APPROVED** [review]

File: crypto/src/signature.rs [hunk 3]
```diff
@@ -571,6 +572,7 @@ mod tests {
     }
 
     #[test]
+    #[cfg(feature = "std")]
```
Comment (mversic) [inline crypto/src/signature.rs:575]: can tests even be run with `no-std`?

File: .github/workflows/iroha2-pr.yml [hunk 4]
```diff
@@ -8,46 +8,131 @@ env:
...
+          rm -f ~/.gradle/caches/modules-2/gc.properties
+
+  long-tests:
+    runs-on: [self-hosted, Linux] #ubuntu-latest
+    container:
+      image: 7272721/i2-ci:latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run long tests
+        run: cargo test --workspace --no-fail-fast -- --ignored --test-threads=1 long
```
Comment (mversic) [inline .github/workflows/iroha2-pr.yml:138]: these tests are prefixed `long`?

File: .github/workflows/iroha2-pr.yml [hunk 5]
```diff
@@ -8,46 +8,131 @@ env:
...
+          chmod +x target/debug/iroha_client_cli
+      - name: Setup test environment
+        run: bash -c './scripts/test_env.sh setup'
+      - name: Genesis test
+        run: bash -c './scripts/tests/genesis.sh || (cat test/peers/iroha0.log; false )'
+      - name: Basic register and mint
+        run: bash -c './scripts/tests/register_mint_quantity.sh || (cat test/peers/iroha0.log; false )'
+        if: always()
+      - name: Cleanup test environment
+        run: bash -c './scripts/test_env.sh cleanup'
```
Comment (mversic) [inline .github/workflows/iroha2-pr.yml:36]: always?

**Review by mversic — APPROVED** [review]

File: crypto/src/signature.rs (hunk 3 above)
Comment (appetrosyan) [reply crypto/src/signature.rs:575]: Some. Not all.

File: .github/workflows/iroha2-pr.yml (hunk 4 above)
Comment (appetrosyan) [reply .github/workflows/iroha2-pr.yml:138]: Not exactly, but it's good enough, until we introduce a better testing fixture.

File: .github/workflows/iroha2-pr.yml (hunk 5 above)
Comment (appetrosyan) [reply .github/workflows/iroha2-pr.yml:36]: Since this doesn't cache on fail, it would be redundant.

**Review by s8sato — APPROVED** [review]

File: .github/workflows/iroha2.yml [hunk 6]
```diff
@@ -1,4 +1,4 @@
-name: Iroha 2 main branch workflow
+name: I2::Release::Deploy
```
Comment (mversic) [inline .github/workflows/iroha2.yml:1]: link-time-optimization could be considered for release builds

Comment (mversic) [reply .github/workflows/iroha2.yml:1]: lto will most likely make noticeable difference in binary size and performance

**Pull request merged**